### PR TITLE
feat(app) add Parameters tab to Protocol Details page

### PR DESCRIPTION
**Pull request by koji**
# Overview
close AUTH-107

Screenshot 2024-03-14 at 5 47 16 PM

- Add `Parameters` tab
- Modify type names that are related to RunTimeParameters
- Fix small design bugs

# Test Plan

# Changelog
- add Parameters tag component and its test
- update a few components to fix small design bugs
- update types

# Review requests

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] merge edge to fix check-js

File: app/src/organisms/ProtocolDetails/ProtocolParameters.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  runTimeParameters,
+}: ProtocolParameterItemsProps): JSX.Element {
+  const { t } = useTranslation('protocol_details')
+
+  const formattedValue = (runTimeParameter: RunTimeParameter): string => {
+    const { type, default: defaultValue } = runTimeParameter
+    switch (type) {
+      case 'int':
+      case 'float':
+        return defaultValue.toString()
```
Comment (jerader) [inline app/src/organisms/ProtocolDetails/ProtocolParameters.tsx]: what about the suffix? Don't we need to add it to the end of the int or float?

File: app/src/organisms/ProtocolDetails/ProtocolParameters.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+    switch (type) {
+      case 'int':
+      case 'float':
+        return minMax
+      case 'boolean':
+        return t('on_off')
+      case 'str':
+        if (defaultValue === 'left' || defaultValue === 'right') {
+          return t('left_right')
```
Comment (jerader) [inline app/src/organisms/ProtocolDetails/ProtocolParameters.tsx]: this is a very special case, i know its in the designs but lets confirm we should special case this? like how often will this param actually be used by customers?

File: app/src/organisms/ProtocolDetails/ProtocolParameters.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+              </StyledTableCell>
+            </StyledTableRow>
+          )
+        })}
+      </tbody>
+    </StyledTable>
+  )
+}
+
+function NoParameter(): JSX.Element {
```
Comment (jerader) [inline app/src/organisms/ProtocolDetails/ProtocolParameters.tsx]: should this be in a separate file?

File: app/src/pages/Protocols/hooks/index.ts [hunk 4]
```diff
@@ -192,15 +192,15 @@ export const useRequiredProtocolHardwareFromAnalysis = (
 
 export const useRunTimeParameters = (
   protocolId: string
-): RunTimeParameters[] => {
+): RunTimeParameter[] => {
```
Comment (jerader) [inline app/src/pages/Protocols/hooks/index.ts:195]: nice thanks for fixing this!

**Review by jerader — COMMENTED** [review]
left some initial feedback. looks good, just want to confirm the very special casing of Left, Right

File: app/src/organisms/ProtocolDetails/ProtocolParameters.tsx (hunk 1 above)
Comment (koji) [reply app/src/organisms/ProtocolDetails/ProtocolParameters.tsx]: forgot to add that. thanks!

**Review by jerader — APPROVED** [review]
lgtm thank you! Once this merges, i'll rebase and fix my ODD protocol details pr

**Pull request merged**